### PR TITLE
More NodeId pruning

### PR DESCRIPTION
**Pull request by ljedrz**
Just another round of the `HirId`ification initiative.

r? @Zoxc

### CODE REVIEW

**Comment by ljedrz** [comment]
Oof, seems like I missed a test.

**Comment by ljedrz** [comment]
Rebased; I wouldn't mind a small priority bump if it's ok due to the number of files that this PR changes.

File: src/librustc_save_analysis/lib.rs [hunk 1]
```diff
@@ -605,8 +606,8 @@ impl<'l, 'tcx> SaveContext<'l, 'tcx> {
         }
     }
 
-    pub fn get_path_res(&self, id: NodeId) -> Res {
-        match self.tcx.hir().get(id) {
+    pub fn get_path_res(&self, hir_id: hir::HirId) -> Res {
```
Comment (Zoxc) [inline src/librustc_save_analysis/lib.rs]: Not sure changing this to a `HirId` is  a good idea. It seems to just add more conversion code.

Comment (ljedrz) [reply src/librustc_save_analysis/lib.rs]: The `NodeId` eventually gets converted to a `HirId` with the call to `get` anyway; my end goal is to convert a bigger part of `librustc_save_analysis` to use `HirId` and remove the conversions while I'm at it.

Comment (Zoxc) [reply src/librustc_save_analysis/lib.rs]: Save analysis uses the AST which means it has to use node ids. So I think you should just change this back.

**Comment by ljedrz** [comment]
@Zoxc Thanks for the review! Comment addressed.

**Pull request merged**